fhevm/params.go

```diff
@@ -40,33 +40,66 @@ type FhevmParams struct {
 }
 
 type GasCosts struct {
-	FheCast             uint64
-	FhePubKey           uint64
-	FheAddSub           map[tfhe.FheUintType]uint64
-	FheDecrypt          map[tfhe.FheUintType]uint64
-	FheBitwiseOp        map[tfhe.FheUintType]uint64
-	FheMul              map[tfhe.FheUintType]uint64
-	FheScalarMul        map[tfhe.FheUintType]uint64
-	FheScalarDiv        map[tfhe.FheUintType]uint64
-	FheScalarRem        map[tfhe.FheUintType]uint64
-	FheShift            map[tfhe.FheUintType]uint64
-	FheScalarShift      map[tfhe.FheUintType]uint64
-	FheEq               map[tfhe.FheUintType]uint64
-	FheLe               map[tfhe.FheUintType]uint64
-	FheMinMax           map[tfhe.FheUintType]uint64
-	FheScalarMinMax     map[tfhe.FheUintType]uint64
-	FheNot              map[tfhe.FheUintType]uint64
-	FheNeg              map[tfhe.FheUintType]uint64
-	FheReencrypt        map[tfhe.FheUintType]uint64
-	FheTrivialEncrypt   map[tfhe.FheUintType]uint64
-	FheRand             map[tfhe.FheUintType]uint64
-	FheIfThenElse       map[tfhe.FheUintType]uint64
-	FheVerify           map[tfhe.FheUintType]uint64
-	FheGetCiphertext    map[tfhe.FheUintType]uint64
+	FheCast           uint64
+	FhePubKey         uint64
+	FheAddSub         map[tfhe.FheUintType]uint64
+	FheDecrypt        map[tfhe.FheUintType]uint64
+	FheBitwiseOp      map[tfhe.FheUintType]uint64
+	FheMul            map[tfhe.FheUintType]uint64
+	FheScalarMul      map[tfhe.FheUintType]uint64
+	FheScalarDiv      map[tfhe.FheUintType]uint64
+	FheScalarRem      map[tfhe.FheUintType]uint64
+	FheShift          map[tfhe.FheUintType]uint64
+	FheScalarShift    map[tfhe.FheUintType]uint64
+	FheEq             map[tfhe.FheUintType]uint64
+	FheLe             map[tfhe.FheUintType]uint64
+	FheMinMax         map[tfhe.FheUintType]uint64
+	FheScalarMinMax   map[tfhe.FheUintType]uint64
+	FheNot            map[tfhe.FheUintType]uint64
+	FheNeg            map[tfhe.FheUintType]uint64
+	FheReencrypt      map[tfhe.FheUintType]uint64
+	FheTrivialEncrypt map[tfhe.FheUintType]uint64
+	FheRand           map[tfhe.FheUintType]uint64
+	FheIfThenElse     map[tfhe.FheUintType]uint64
+	FheVerify         map[tfhe.FheUintType]uint64
+	FheGetCiphertext  map[tfhe.FheUintType]uint64
+	SgxAddSub         map[tfhe.FheUintType]uint64
+	SgxMul            map[tfhe.FheUintType]uint64
+	SgxEncrypt        map[tfhe.FheUintType]uint64
+	SgxDecrypt        map[tfhe.FheUintType]uint64
 }
 
 func DefaultGasCosts() GasCosts {
 	return GasCosts{
+		SgxAddSub: map[tfhe.FheUintType]uint64{
+			tfhe.FheUint4:  550,
```

Comment:
Looking at these values, let's actually make it 1000x less than FHE!

For exampe, 55 gas for an addition seems reasonable to me (it's ~3 gas I think for a normal solidity addition)